test/testAllRenders.ts

```diff
@@ -0,0 +1,48 @@
+import * as mwoffliner from '../src/mwoffliner.lib.js'
+import { execa } from 'execa'
+import { RENDERERS_LIST } from '../src/util/const.js'
+import { zimcheckAvailable, zimdumpAvailable } from './util.js'
+
+/*
+  This is the template for e2e tests of different wikis
+  1. Verify zimcheck and zimdump availability
+  2. Gets output file and checks its integrity
+  3. Returns output file per renderer in the callback function
+*/
+
+async function getOutFiles(renderName: string, testId: string, articleList: string, mwUrl: string, format?: string | string[]): Promise<any> {
+  const parameters = {
+    mwUrl,
+    adminEmail: 'test@kiwix.org',
+    outputDirectory: testId,
+    redis: process.env.REDIS,
+    articleList,
+    forceRender: renderName,
+    format,
+  }
+
+  await execa('redis-cli flushall', { shell: true })
+  const outFiles = await mwoffliner.execute(parameters)
+
+  return outFiles
+}
+
+export async function testAllRenders(mwUrl: string, articleList: string, format: string | string[], callback) {
+  const zimcheckIsAvailable = await zimcheckAvailable()
+  const zimdumpIsAvailable = await zimdumpAvailable()
+
+  if (!zimcheckIsAvailable || !zimdumpIsAvailable) {
```

Comment:
You can assume these tools are there.

Reply:
I updated this condition to exit if no zimtool available (if I understood your comment correctly here).

Reply:
Not really I guess, you test that once at the whole test process:
* either it is there and you can assume in all the tests, they are there, so you don't test the presence of zimcheck and zimdump anymore
* or you just exit with an error

Reply:
I added lazy **checkZimTools()**